File: test-suite/src/test/java/io/micronaut/microstream/docs/CacheTest.java
package io.micronaut.microstream.docs;

import io.micronaut.context.ApplicationContext;
import io.micronaut.core.util.CollectionUtils;
import io.micronaut.runtime.server.EmbeddedServer;
import org.junit.jupiter.api.Test;

import java.util.Map;
import java.util.UUID;

import static org.junit.jupiter.api.Assertions.assertEquals;

class CacheTest {

    @Test
    void cacheWorksAsExpected() {
        Map<String, Object> config = CollectionUtils.mapOf(
            "storageDirectory", "build/microstream-cache-" + UUID.randomUUID()
        );
        // When we create the app, and use a cached method
        try (EmbeddedServer server = ApplicationContext.run(EmbeddedServer.class, config, "cache")) {
            CounterService counter = server.getApplicationContext().getBean(CounterService.class);
            counter.setCount("Tim", 1337L);
            Long count = counter.currentCount("Tim");
            assertEquals(1337L, count);

            // Change the store so we check we're seeing a cached value
            counter.counters.put("Tim", -1L);
            count = counter.currentCount("Tim");
            assertEquals(1337L, count);
        }
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/microstream/docs/CacheTest.kt
package io.micronaut.microstream.docs

import io.micronaut.context.ApplicationContext
import io.micronaut.runtime.server.EmbeddedServer
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import java.util.UUID

class CacheTest {

    @Test
    fun cacheWorksAsExpected() {
        val config = mapOf("storageDirectory" to "build/microstream-cache-${UUID.randomUUID()}")
        ApplicationContext.run(EmbeddedServer::class.java, config, "cache").use {
            val counter = it.applicationContext.getBean(CounterService::class.java)
            counter.setCount("Tim", 1337)

            var currentCount = counter.currentCount("Tim")
            assertEquals(1337, currentCount)

            // Change the value in the map to check we are using the cache
            counter.counters["Tim"] = 42

            currentCount = counter.currentCount("Tim")
            assertEquals(1337, currentCount)
        }
    }
}
